test/assignment-5/CollateralizedVault.t.sol

```diff
@@ -0,0 +1,270 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "forge-std/Test.sol";
+import {CollateralizedVault} from "src/assignment-5/CollateralizedVault.sol";
+import {Dai} from "src/assignment-5/DAI.sol";
+import {WETH9} from "src/assignment-5/WETH.sol";
+import {IERC20} from "yield-utils-v2/token/IERC20.sol";
+import {ChainlinkPriceFeedMock} from "src/assignment-5/ChainlinkPriceFeedMock.sol";
+
+abstract contract ZeroState is Test {
+
+    using stdStorage for StdStorage;
+
+    address USER = address(1);
+    CollateralizedVault vault;
+    Dai dai;
+    WETH9 weth;
+    ChainlinkPriceFeedMock priceFeedMock;
+
+    function setUp() public virtual {
+        dai = new Dai(block.chainid);
+        weth = new WETH9();
+        priceFeedMock = new ChainlinkPriceFeedMock();
+        priceFeedMock.setPrice(500000000000000); // = 1/2000
+        vault = new CollateralizedVault(address(dai), address(weth), address(priceFeedMock));
+
+        setDaiBalance(address(vault), 10000 ether);
+        setWethBalance(USER, 10 ether);
+        vm.prank(USER);
+        weth.approve(address(vault), 10 ether);
+    }
+
+    function setDaiBalance(address dst, uint256 balance) public {
+        stdstore
```

Comment:
love to see the use of stdstore --

curious, do u think theres an advantage to doing it this way instead just using the standard way of `mint()`ing to the address?

anyways, i fully support all of us playing around w stdstore as much as possible to find better patterns

Reply:
@devtooligan it was part of the assignment!

> Using mock WETH and DAI contracts gets you a long way, but you can't always do that. Remove the mint() function and use the foundry cheatcodes to directly edit the registry that stores your balance of each.

Reply:
oh sick, i didnt know they added that.

so after using it how do u feel?  is there any advantage to doing this?

Reply:
the advantage is that I don't need to patch an external contract